packages/product/integration-tests/__tests__/services/product-module-service/product-variants.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { TestDatabase } from "../../../utils"
+import { setupModuleService } from "../../../utils"
+import { ProductModuleService } from "@services"
+import { Product, ProductVariant } from "@models"
+import { SqlEntityManager } from "@mikro-orm/postgresql"
+import { ProductTypes } from "@medusajs/types"
+
+describe("ProductModuleService product variants", () => {
+  let service: ProductModuleService
+  let testManager: SqlEntityManager
+  let repositoryManager: SqlEntityManager
+  let variantOne: ProductVariant
+  let variantTwo: ProductVariant
+  let productOne: Product
+  let productTwo: Product
+
+  beforeEach(async () => {
+    await TestDatabase.setupDatabase()
+    repositoryManager = await TestDatabase.forkManager()
+    service = setupModuleService({ repositoryManager })
+    testManager = await TestDatabase.forkManager()
+
+    productOne = testManager.create(Product, {
+      id: "product-1",
+      title: "product 1",
+      status: ProductTypes.ProductStatus.PUBLISHED,
+    })
+
+    productTwo = testManager.create(Product, {
+      id: "product-2",
+      title: "product 2",
+      status: ProductTypes.ProductStatus.PUBLISHED,
+    })
+
+    variantOne = testManager.create(ProductVariant, {
+      id: "test-1",
+      title: "variant 1",
+      inventory_quantity: 10,
+      product: productOne,
+    })
+
+    variantTwo = testManager.create(ProductVariant, {
+      id: "test-2",
+      title: "variant",
+      inventory_quantity: 10,
+      product: productTwo,
+    })
+
+    await testManager.persistAndFlush([variantOne, variantTwo])
+  })
+
+  afterEach(async () => {
+    await TestDatabase.clearDatabase()
+  })
+
+  describe("listAndCountVariants", () => {
+    it("selecting by properties, scopes out the results", async () => {
```

Comment:
todo: `should return the variants and count` I don't see the properties, scoped out of the result. Can I get you to apply the same approach to all the tests description as we previously discussed. I will enforce more convention to the tests as it becomes quite difficult to retrieve and read the description which does not really describe the test purpose 🚀